### PR TITLE
`vde: defer vnl validation until starting qemu` + `pkg/start: exit when hostagent process exited`

### LINKED ISSUE #146: Do not validate the `network.vde[*].vnl`
The VNL should only be checked/verified when the instance is started.

Otherwise, if the vde_switch has stopped/crashed, all the instances using this VNL become unaccessible to Lima and can't even be deleted without manually editing the `lima.yaml` file (or restarting the `vde_switch`):

```console
$ limactl ls
ERRO[0000] instance "std" does not exist?                error="field `network.vde[0].vnl` \"/var/run/vde.ctl\" failed stat: stat /var/run/vde.ctl: no such file or directory"
NAME    STATUS    SSH    ARCH    DIR
$ limactl delete -f std
WARN[0000] Ignoring non-existent instance "std"
```

The `delete` command doesn't remove the instance directory because the config file no longer validates.

### PR DESCRIPTION
**Pull request by AkihiroSuda**
Fix #146

### CODE REVIEW

**Comment by jandubois** [comment]
This did not work for me when I tested it with `sudo make uninstall` in the `vde_vmnet` directory because the uninstall leaves the `/var/run/vde.ctl` directory behind, it just removes the sockets from inside the directory:

```console
$ ls -l /var/run/vde.ctl/
total 0
srwxrwx---  1 daemon  staff  0  6 Aug 12:19 001.5
srw-rw----  1 daemon  staff  0  6 Aug 12:19 ctl
$ sudo make uninstall
launchctl unload -w "/Library/LaunchDaemons/io.github.lima-vm.vde_vmnet.plist"
/Library/LaunchDaemons/io.github.lima-vm.vde_vmnet.plist: Operation now in progress
launchctl unload -w "/Library/LaunchDaemons/io.github.virtualsquare.vde-2.vde_switch.plist"
/Library/LaunchDaemons/io.github.virtualsquare.vde-2.vde_switch.plist: Operation now in progress
rm -f "/Library/LaunchDaemons/io.github.lima-vm.vde_vmnet.plist"
rm -f "/Library/LaunchDaemons/io.github.virtualsquare.vde-2.vde_switch.plist"
rm -f "//usr/local/bin/vde_vmnet"
$ ls -l /var/run/vde.ctl/
$ limactl ls
ERRO[0000] instance "std" does not exist?                error="field `network.vde[0].vnl` control socket \"/var/run/vde.ctl/ctl\" failed stat: stat /var/run/vde.ctl/ctl: no such file or directory"
NAME    STATUS    SSH    ARCH    DIR
```

**Comment by AkihiroSuda** [comment]
Updated, sorry for delay.

I'll release v0.6.1 after merging this.

**Review by jandubois — CHANGES_REQUESTED** [review]
Starting a new instance with `example/vmnet.yaml` config while vde_switch is not running hangs:

```console
$ l start examples/vmnet.yaml
? Creating an instance "vmnet" Proceed with the default configuration
INFO[0001] Downloading "https://github.com/containerd/nerdctl/releases/download/v0.11.0/nerdctl-full-0.11.0-linux-amd64.tar.gz" (sha256:a491a3129beddf2feb41a8ea9dcd0d3e4c9cb36c3a21046675e425f4b3daa5a9)
INFO[0002] Using cache "/Users/jan/Library/Caches/lima/download/by-url-sha256/fdbae42d5655d51afaedd0563a278bc93ea15d4f90aa8722164682cb77fff619/data"
INFO[0003] Attempting to download the image from "https://cloud-images.ubuntu.com/hirsute/current/hirsute-server-cloudimg-amd64.img"
INFO[0005] Using cache "/Users/jan/Library/Caches/lima/download/by-url-sha256/e1fed960ebd29619676c7ab7535bc83f7fb2ad71739edb6fde4e17bce0b61a47/data"
INFO[0005] [hostagent] time="2021-08-19T23:00:53-07:00" level=fatal msg="stat /var/run/vde.ctl: no such file or directory"
```

**Comment by AkihiroSuda** [comment]
@jandubois 
Fixed in the second commit `pkg/start: exit when hostagent process exited`

**Comment by AkihiroSuda** [comment]
Added another commit `pkg/qemu: check features before starting QEMU`.

@jandubois PTAL, I'll release v0.6.1 after merging this.

File: pkg/qemu/qemu.go [hunk 1]
```diff
@@ -197,11 +247,19 @@ func Cmdline(cfg Config) (string, []string, error) {
...
+	if len(y.Network.VDE) > 0 && !strings.Contains(string(features.NetdevHelp), "vde") {
+		return "", nil, fmt.Errorf("netdev \"vde\" is not supported by %s ( Hint: recompile QEMU with `configure --enable-vde` )", exe)
+	}
 	for i, vde := range y.Network.VDE {
 		// VDE4 accepts VNL like vde:///var/run/vde.ctl as well as file path like /var/run/vde.ctl .
 		// VDE2 only accepts the latter form.
 		// VDE2 supports macOS but VDE4 does not yet, so we trim vde:// prefix here for VDE2 compatibility.
 		vdeSock := strings.TrimPrefix(vde.VNL, "vde://")
+		if !strings.Contains(vdeSock, "://") {
+			if _, err := os.Stat(vdeSock); err != nil {
```
[the remark this answers is not in the document]
Comment (AkihiroSuda) [reply pkg/qemu/qemu.go:265]: updated

**Comment by AkihiroSuda** [comment]
@jandubois I'll release v0.6.1 after merging this

**Review by jandubois — APPROVED** [review]
I think it is fine now.

I just wish `limactl` would display some status when the network has been broken. But that should be a separate PR.

**Pull request merged**